rs/sns/testing/README.md

```diff
@@ -44,14 +43,183 @@ To run the scenario on the local PocketIC instance:
+
+Make sure to use `sns-testing` identity when creating the proposal.
+
+<details>
+<summary>Sample SNS proposal creation workflow</summary>
+<br>
+
+The example will use `//rs/sns/testing:sns_testing_canister` canister as SNS-controlled canister and will base on the [init YAML file from SNS CLI](../cli/test_sns_init_v2.yaml).
+
+**Make sure to use `start_time: null` in the swap parameters to ensure that the swap starts right away after the NNS proposal is executed.**
+
+1) Build and deploy `test` canister:
+   ```
+   bazel build //rs/sns/testing:sns_testing_canister
+   # 'r7inp-6aaaa-aaaaa-aaabq-cai' is the NNS Root canister
+   dfx canister --network http://127.0.0.1:8080 --identity sns-testing create test --controller r7inp-6aaaa-aaaaa-aaabq-cai --controller sns-testing --no-wallet
+   dfx canister --network http://127.0.0.1:8080 --identity sns-testing install test --wasm "$(bazel info bazel-bin)/rs/sns/testing/sns_testing_canister.wasm.gz"
+   ```
+
+2) Adjust init YAML file (you will need [`yq`](https://github.com/mikefarah/yq) to be installed to do this):
+   ```
+   yq -i ".dapp_canisters |= [\""$(dfx canister --network http://127.0.0.1:8080 id test)"\"]" ../cli/test_sns_init_v2.yaml
+   yq -i ".Distribution.Neurons[0].principal |= \""$(dfx identity get-principal --identity sns-testing)"\"" ../cli/test_sns_init_v2.yaml
+   yq -i ".Swap.start_time |= null" ../cli/test_sns_init_v2.yaml
+   ```
+3) Propose to create the new SNS:
+   ```
+   pushd ../cli
+   # //rs/sns/cli:sns doesn't support CLI-provided identities despite '--identity' option
+   dfx identity use sns-testing
+   bazel run //rs/sns/cli:sns -- propose --network http://127.0.0.1:8080 --neuron-id 1 $PWD/test_sns_init_v2.yaml
+   popd
+   ```
+
+4) Complete the swap for the newly created SNS
+   ```
+   SNS_NAME="$(yq -r .name ../cli/test_sns_init_v2.yaml)"
+   bazel run //rs/sns/testing:sns-testing -- --network http://127.0.0.1:8080 swap-complete --sns-name "$SNS_NAME"
+   ```
+</details>
+
+Once the NNS proposal to create the new SNS is adopted and executed, the SNS swap is supposed to open.
+Use `bazel run //rs/sns/testing:cli -- run swap-complete` to generate swap participations and complete the swap:
+```
+SNS_NAME="$(yq -r .name <Path to SNS YAML>)"
+bazel run //rs/sns/testing:sns-testing -- --network http://127.0.0.1:8080 swap-complete --sns-name "$SNS_NAME" --follow-principal-neurons "$(dfx identity get-principal --identity sns-testing)"
+```
+
+This command will generate required number of participations with the sufficient amount of direct participants to complete the swap.
+
+This command has optional `--follow-principal-neurons` and `--follow-neuron` arguments that accept `PrincipalId` and `NeuronId` to make swap
+participant neurons follow given neurons.
+
+### SNS voting
+
+The newly created SNS is now fully functional and is ready for proposals.
```

Comment:
Actually, it's also possible to send some proposals during the swap, not only after it completes

Reply:
Oh, I didn't know that 😅